test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -193,18 +199,34 @@ void performFailoverTest() throws InterruptedException {
                     } else {
                         log.debug("Read timeout");
                     }
-                } catch (ConnectionClosedException e) {
+                } catch (RetriesExhaustedException e) {
                     log.warn("Test exception in reading events: ", e);
                     continue;
-                } catch (ReinitializationRequiredException e) {
+                } catch (Throwable e) {
+                    if (e.getCause() instanceof RetriesExhaustedException) {
+                        log.warn("Test exception in reading events: ", e);
+                        continue;
```

Comment:
@shrids I think @tkaitchuck is saying something different. He is saying that we should not ignore and proceed. Even though it is harmless in our case to proceed, the retries exhaustion is not really expected in this test case, so we better fail as soon and investigate with fewer log messages to inspect. 

I can go either way, since our main goal here is to make sure we read as much as we write, but I agree that the retries exhaustion is unexpected and undesirable, so if we keep hitting it, we should investigate.